native/optflow.cc

```diff
@@ -0,0 +1,126 @@
+                                 int numIters,
+                                 int polyN,
+                                 double polySigma,
+                                 int flags) {
+    auto array_from = cv::InputArray(*from);
+    auto array_to = cv::InputArray(*to);
+    auto array_out = cv::InputOutputArray(*out);
+
+    auto optical_flow =
+        cv::FarnebackOpticalFlow::create(numLevels, pyrScale, fastPyramids, winSize, numIters, polyN, polySigma, flags);
+
+    optical_flow->calc(array_from, array_to, array_out);
+}
+
+void calc_optical_flow_dtvl1(cv::Mat* from,
+                             cv::Mat* to,
+                             cv::Mat* out,
+                             double tau,
+                             double lambda,
+                             double theta,
+                             int nscales,
+                             int warps,
+                             double epsilon,
+                             int innerIterations,
+                             int outerIterations,
+                             double scaleStep,
+                             double gamma,
+                             int medianFiltering,
+                             bool useInitialFlow) {
+    auto array_from = cv::InputArray(*from);
+    auto array_to = cv::InputArray(*to);
+    auto array_out = cv::InputOutputArray(*out);
+
+    auto optical_flow = cv::DualTVL1OpticalFlow::create(tau,
+                                                        lambda,
+                                                        theta,
+                                                        nscales,
+                                                        warps,
+                                                        epsilon,
+                                                        innerIterations,
+                                                        outerIterations,
+                                                        scaleStep,
+                                                        gamma,
+                                                        medianFiltering,
+                                                        useInitialFlow);
+
+    optical_flow->calc(array_from, array_to, array_out);
+}
+
+void calc_optical_flow_dis(cv::Mat* from, cv::Mat* to, cv::Mat* out, unsigned int preset) {
+    auto array_from = cv::InputArray(*from);
+    auto array_to = cv::InputArray(*to);
+    auto array_out = cv::InputOutputArray(*out);
+
+    auto optical_flow = cv::optflow::createOptFlow_DIS(preset);
+    optical_flow->calc(array_from, array_to, array_out);
+}
+
+void calc_optical_flow_std(cv::Mat* from, cv::Mat* to, cv::Mat* out) {
+    auto array_from = cv::InputArray(*from);
+    auto array_to = cv::InputArray(*to);
+    auto array_out = cv::InputOutputArray(*out);
+
+    auto optical_flow = cv::optflow::createOptFlow_SparseToDense();
+    optical_flow->calc(array_from, array_to, array_out);
```

Comment:
can these calls be wrapped? these functions could throw exceptions

see my pr [here](https://github.com/nebgnahz/cv-rs/pull/87/files) - using `EmptyResult::FromFunction` or `Result::FromFunction` would be useful so that errors don't panic